File: app/src/main/java/com/holokenmod/creation/MultiplicationCreator.java
package com.holokenmod.creation;

import com.holokenmod.Grid;

import java.util.ArrayList;

public class MultiplicationCreator {
    private final int target_sum;
    private final int n_cells;
    private final Grid grid;
    private final GridCageCreator cageCreator;

    public MultiplicationCreator(GridCageCreator cageCreator, Grid grid, int target_sum, int n_cells) {
        this.cageCreator = cageCreator;
        this.grid = grid;
        this.target_sum = target_sum;
        this.n_cells = n_cells;
    }

    public ArrayList<int[]> create() {
        if (target_sum == 0) {
            return new MultiplicationZeroCreator(
                    cageCreator,
                    grid,
                    n_cells).create();
        }

        return new MultiplicationNonZeroCreator(
                cageCreator,
                grid,
                target_sum,
                n_cells).create();
    }
}

File: holoken-core/src/main/kotlin/com/holokenmod/creation/cage/MultiplicationCreator.kt
package com.holokenmod.creation.cage

import com.holokenmod.grid.Grid

internal class MultiplicationCreator(
    private val cageCreator: GridSingleCageCreator,
    private val grid: Grid,
    private val targetSum: Int,
    private val numberOfCells: Int
) {
    fun create(): ArrayList<IntArray> {
        return if (targetSum == 0) {
            MultiplicationZeroCreator(
                cageCreator,
                grid,
                numberOfCells
            ).create()
        } else {
            MultiplicationNonZeroCreator(
                cageCreator,
                grid,
                targetSum,
                numberOfCells
            ).create()
        }
    }
}
